mllib/src/main/scala/org/apache/spark/mllib/feature/IDF.scala

```diff
@@ -218,7 +218,7 @@ private object IDFModel {
           newValues(k) = values(k) * idf(indices(k))
           k += 1
         }
-        Vectors.sparse(n, indices, newValues)
+        Vectors.sparse(n, indices, newValues).toSparse
```

Comment:
If the `Array` version of the method is commonly used because it contains some additional conveniences and safety checks, that would align with @srowen's point that the expectations may be different for a factory method. If that's what's driving the preferred usage of the `Array` version, it would tip me toward filtering out explicit zeros there (prior to the sort).